lib/core/src/recover/handlers/tests/handle_chain_receive_swap_tests_integration.rs

```diff
@@ -0,0 +1,601 @@
+#[cfg(test)]
+mod test {
+    use crate::{
+        chain::liquid::MockLiquidChainService,
+        model::{ChainSwap, PaymentState, SwapMetadata},
+        recover::{
+            handlers::ChainReceiveSwapHandler,
+            model::{HistoryTxId, RecoveryContext, TxMap},
+        },
+        swapper::MockSwapper,
+    };
+    use boltz_client::{
+        bitcoin::{self, transaction::Version, Sequence},
+        Amount, LockTime,
+    };
+    use electrum_client::GetBalanceRes;
+    use lwk_wollet::{
+        elements::{self, AssetId, Transaction, TxIn, TxInWitness, Txid},
+        elements_miniscript::slip77::MasterBlindingKey,
+        WalletTx,
+    };
+    use std::{
+        collections::{BTreeMap, HashMap},
+        str::FromStr,
+        sync::Arc,
+    };
+
+    #[tokio::test]
```

Comment:
These tests could be adapted wasm also, you can:

- Replace all `#[tokio::test]` with `#[sdk_macros::async_test_all]`
- Replace all `#[test]` with `#[sdk_macros::test_all]`
- Add into each test module
```
    #[cfg(all(target_family = "wasm", target_os = "unknown"))]
    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
```

Reply:
Done